batch/src/main/scala/uk/gov/ons/addressindex/models/EsDocument.scala

```diff
@@ -28,8 +28,12 @@ abstract class EsDocument {
     val premsAndThoroughfare = (premises, thoroughfares) match {
       case (sub_build :: build :: Some(number) :: Nil, Some(thorough_first) :: thorough_rest) =>
         sub_build :: build :: poBox :: Some(number + " " + thorough_first) :: thorough_rest
+      case (sub_build :: build :: Some(number) :: Nil, None :: thorough_rest) =>
+        sub_build :: build :: poBox :: Some(number + " " + thorough_rest.headOption.getOrElse(Some("")).getOrElse("")) :: Nil
```

Comment:
This doesn't look right. I'm assuming all you want is to detect the empty list case, which you can do so:
```scala
case (sub_build :: build :: Some(number) :: Nil, Nil) =>
  sub_build :: build :: poBox :: Some(number + " " + "") :: Nil
```
If that looks right? Naturally the `+ ""` can be removed. It also demonstrates that it might leave extra spaces with nothing following.